### DISCUSSION TITLE
DEPRECATION:  replace `--scanners config` with `--scanners misconfig`

### DISCUSSION THREAD

**Discussion by knqyf263** [aquasecurity/trivy#5586] [Announcements]
## Summary
Please migrate `--scanners config` to `--scanners misconfig`

### Before

```
$ trivy fs --scanners config /path/to/repo
```

### After

```
$ trivy fs --scanners misconfig /path/to/repo
```

## Description
To enhance clarity and usability, the flag `--scanners config` is being renamed to `--scanners misconfig`. The new flag, `--scanners misconfig`, more accurately reflects its function of detecting misconfigurations in configurations.

To facilitate a smooth transition, the `--scanners config` flag will remain operational for a certain period. However, the recommendation is to adopt the new `--scanners misconfig` flag in upcoming scans.

**Comment by mattduguid** [comment]
there are a number of open issues on this change some well over a year old, 
https://github.com/aquasecurity/trivy-azure-pipelines-task/issues/41
https://github.com/aquasecurity/trivy-azure-pipelines-task/issues/36
...etc...

we've been trying to use the azure devops task but we always get the same warning,
```bash
2024/07/02 23:04:06 WARN '--security-checks' is deprecated. Use '--scanners' instead.
```

we tried this workaround https://github.com/aquasecurity/trivy-azure-pipelines-task/issues/41#issuecomment-1609556260 but no luck there either, we pass in what the comment suggests but the error persists,
```yaml
 options: '--config /tmp/trivy.yaml'
 ```
 
if we use an inline script task (not the trivy task) in azure devops pipelines, we can download trivy and execute it that way as a workaround, but the market place task is no good as it generates a WARN which is enough for it to error
